Modules/Loadable/Markups/MRML/vtkMRMLMarkupsCurveNode.cxx

```diff
@@ -1233,7 +1233,8 @@ void vtkMRMLMarkupsCurveNode::ProcessMRMLEvents(vtkObject* caller,
 //---------------------------------------------------------------------------
 void vtkMRMLMarkupsCurveNode::OnNodeReferenceAdded(vtkMRMLNodeReference* reference)
 {
-  if (strcmp(reference->GetReferenceRole(), this->GetShortestDistanceSurfaceNodeReferenceRole()) == 0)
+  if (strcmp(reference->GetReferenceRole(), this->GetShortestDistanceSurfaceNodeReferenceRole()) == 0 ||
+      strcmp(reference->GetReferenceRole(), this->TransformNodeReferenceRole) == 0)
```

Comment:
TransformModifiedEvent is not being called, or at least ProcessMRMLEvents is not receiving it.

Reply:
This is odd. The event should be invoked from here: https://github.com/Slicer/Slicer/blob/9ea355344f1b8107ef4f4b64630548b12cfbc948/Libs/MRML/Core/vtkMRMLTransformableNode.cxx#L284

Do we get TransformModifiedEvent when the transform changes but not when it is applied to the transformable node?

Reply:
Yes, TransformModifiedEvent is called if the parent transform is modified, but not when it is applied/removed.

I'll look some more into why it's not being called.

Reply:
Thank you. If this does not work well then it may cause issues elsewhere, too. Have you figured out what's happening?

Reply:
Yes, it's simply that the node doesn't attach an observer to itself, so it ProcessMRMLEvents doesn't get called if the transform node is added/removed.

Adding an observer in the node to itself fixes the problem, but I don't know if that's a good solution.

Reply:
Finally I had time to review this and it is all good.

I was just worried that observing transform reference may be be missed somewhere, but this was just a special case because normally TransformModified event is observed from outside (which captures all transform reference and transform content changes) and here we don't use that because we don't observe ourselves.

As you wrote, attaching an observer to ourselves could work, too. However, probably it is simpler and safer to not add an extra observer.